ui/src/app/machines/views/MachineList/MachineListHeader/ActionFormWrapper/DeployForm/DeployFormFields/DeployFormFields.tsx

```diff
@@ -102,6 +103,30 @@ export const DeployFormFields = (): JSX.Element => {
             ]}
             inline
           />
+          <FormikField
+            label="Cloud-init user data..."
+            name="toggleUserData"
+            type="checkbox"
+            onChange={(evt: React.ChangeEvent<HTMLInputElement>) => {
+              handleChange(evt);
+              setUserDataVisible(evt.target.checked);
+              setFieldValue("userData", "");
```

Comment:
I think clearing the value should only happen on submit if the "Customise" checkbox isn't checked, otherwise it's kind of a shame to lose the data if you uncheck it without meaning to. I believe that's the intended behaviour (possibly only discussed at a standup, sorry!)

Reply:
Done